src/twisted/test/test_udp.py

```diff
@@ -563,10 +586,45 @@ def flushErrors(ignored):
         return finished
 
 
+def checkMulticastGroups() -> tuple[bool, bool]:
+    s = socket(AF_INET, SOCK_DGRAM)
+    s.bind(("0.0.0.0", 0))
+    try:
+        s.setsockopt(
+            IPPROTO_IP,
+            IP_ADD_MEMBERSHIP,
+            inet_pton(AF_INET, "225.0.0.250") + inet_pton(AF_INET, "0.0.0.0"),
+        )
+    except OSError:  # pragma: no cover
+        ipv4Support = False  # pragma: no cover
```

Comment:
I think that a signele `# pragma: no cover is enough` as it will include all the code from that block
maybe add a comment about the assumption that is made here

```suggestion
    except OSError:  # pragma: no cover
        # Something went wrong with joining an arbitrary
        # IPv4 group. Most probably multicast is not supported.
        ipv4Support = False
```